docs/Pitfalls.asciidoc

```diff
@@ -53,3 +53,46 @@ impacted by snapshot creation. Loading VM snapshots can also have an impact on
 SUT behavior as the execution of the first step after loading a snapshot might
 be delayed. This can lead to problems if the executed tests do not foresee an
 appropriate timeout margin.
+
+== DB migration from sqlite to postgreSQL
+In order to start using postgreSQL DBMS with openQA one need to perform following steps:
+- Install postgresql-server.
+- Adjust postgresql configuration file,
+namely `/var/lib/pgsql/data/pg_hba.conf`. For example:
+
+```
+local   all             all                                     md5
+local   all             postgres                                md5
+local   all             openqa                                  md5
+# IPv4 local connections:
+host    all             all             127.0.0.1/32            md5
+# IPv6 local connections:
+host    all             all             ::1/128                 md5
+
+```
+- Connect to the DBMS (`sudo sudo -u postgres psql`) and create user, db and delegate
+permissions to that user:
+
+```
+CREATE DATABASE openqa;
+CREATE USER openqa;
+GRANT ALL PRIVILEGES ON DATABASE openqa to openqa;
+ALTER USER openqa PASSWORD 'nots3cr3t';
+```
+- Restart postgresql DBMS
+- Adjust `/etc/openqa/database.ini` to use postgreSQL database according to the
+installation documentation
+- Init database using initdb script
+
+
+To migrate api keys run following commands (db file is located in `/var/lib/openqa/db`:
+
+```
+# To export data
+sqlite3 db.sqlite -csv -separator ',' 'select * from api_keys;' > apikeys.csv
+# To import
+psql -U postgres -d openqa -c "copy api_keys from '/var/lib/openqa/db/apikeys.csv' with (format csv);"
+```
+
+In case you need to migrate job groups, test suites, use dump_templates and
```

Comment:
\o/

Reply:
[This](https://pgloader.io/howto/sqlite.html) could help too for a full migration, but comes with it's own issues!

Reply:
I guess someone has to try it out, maybe we can find someone who hasn't upgraded openQA yet =)